node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/Rebalancer.java

```diff
@@ -90,20 +107,52 @@ private Move findBestMove(NodeList allNodes) {
                 double skewReductionAtToHost = skewReductionByAdding(node, toHost, capacity);
                 double netSkewReduction = skewReductionAtFromHost + skewReductionAtToHost;
                 if (netSkewReduction > bestMove.netSkewReduction)
-                    bestMove = new Move(node, netSkewReduction);
+                    bestMove = new Move(node, toHost, netSkewReduction);
             }
         }
         return bestMove;
     }
 
-    private void markWantToRetire(Node node) {
+    /** Returns true only if this operation changes the state of the wantToRetire flag */
+    private boolean markWantToRetire(Node node, boolean wantToRetire) {
         try (Mutex lock = nodeRepository().lock(node)) {
             Optional<Node> nodeToMove = nodeRepository().getNode(node.hostname());
-            if (nodeToMove.isEmpty()) return;
-            if (nodeToMove.get().state() != Node.State.active) return;
+            if (nodeToMove.isEmpty()) return false;
+            if (nodeToMove.get().state() != Node.State.active) return false;
+
+            if (node.status().wantToRetire() == wantToRetire) return false;
 
-            nodeRepository().write(nodeToMove.get().withWantToRetire(true, Agent.system, clock.instant()), lock);
-            log.info("Marked " + nodeToMove.get() + " as want to retire to reduce allocation skew");
+            nodeRepository().write(nodeToMove.get().withWantToRetire(wantToRetire, Agent.system, clock.instant()), lock);
+            return true;
+        }
+    }
+
+    /**
+     * Try a redeployment to effect the chosen move.
+     * If it can be done, that's ok; we'll try this or another move later.
+     *
+     * @return true if the move was done, false if it couldn't be
+     */
+    private boolean deployTo(Move move) {
+        ApplicationId application = move.node.allocation().get().owner();
+        try (MaintenanceDeployment deployment = new MaintenanceDeployment(application, deployer, nodeRepository())) {
+            if ( ! deployment.isValid()) return false;
+
+            boolean couldMarkRetiredNow = markWantToRetire(move.node, true);
+            if ( ! couldMarkRetiredNow) return false;
+
+            try {
+                if ( ! deployment.prepare()) return false;
+                if (nodeRepository().getNodes(application, Node.State.reserved).stream().noneMatch(node -> node.hasParent(move.toHost.hostname())))
+                    return false; // Deployment is not moving the from node to the target we identified for some reason
```

Comment:
This leaves the reserved node with the application for 20-27 mins (until it is expired by `ReservationExpirer`), this is especially unfortunate in CD when we want to fail integration tests when some node are stuck in a non-active state.

We should prepare the application again with the node unmarked as wantToRetire or move it directly to dirty if that is safe.

FYI: @hmusum

Reply:
The latter should be fine here. After lunch ...

Reply:
https://github.com/vespa-engine/vespa/pull/11355